zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/DefaultClientChannelManager.java

```diff
@@ -373,6 +385,25 @@ public boolean remove(PooledConnection conn) {
                     null);
         }
 
+        InetSocketAddress serverAddr;
+        try {
+            InetAddress ipAddr = InetAddresses.forString(rawHost);
+            serverAddr = new InetSocketAddress(ipAddr, port);
+        } catch (IllegalArgumentException e1) {
+            LOG.warn("NettyClientConnectionFactory got an unresolved address, addr: {}", rawHost);
+            Counter unresolvedDiscoveryHost = SpectatorUtils.newCounter(
+                    "unresolvedDiscoveryHost",
+                    connPoolConfig.getOriginName() == null ? "unknownOrigin" : connPoolConfig.getOriginName());
+            unresolvedDiscoveryHost.increment();
+            try {
+                serverAddr = new InetSocketAddress(rawHost, port);
```

Comment:
If the intention here is to create a unresolved addr, perhaps use `createUnresolved`? Not sure I understand this, after we know the resolution attempt failed.

Reply:
createUnresolved would allow an unresolved address to propagate down the call chain to NettyClientConnectionFactory, which asserts that should not be done.

As mentioned in the PR description, I plan on rewriting this function to be the point of async name resolution.